src/main/java/com/cloudbees/plugins/deployer/sources/MavenArtifactDeploySource.java

```diff
@@ -158,7 +158,7 @@ public String getGroupId() {
      */
     @NonNull
     public String getType() {
-        return type == null ? DEFAULT_TYPE : type;
+        return type;
```

Comment:
I can be missing something, but type is being set in the constructor, where the null check is already done.